cafs/ipfs/migrate.go

```diff
@@ -2,14 +2,65 @@ package ipfs_filestore
 
 import (
 	"fmt"
+	"io/ioutil"
+	"os"
 
 	fsrepo "github.com/ipfs/go-ipfs/repo/fsrepo"
 	migrate "github.com/ipfs/go-ipfs/repo/fsrepo/migrations"
+	"github.com/otiai10/copy"
 )
 
 // ErrNeedMigration is an
 var ErrNeedMigration = fmt.Errorf(`ipfs: need datastore migration`)
 
+// RunMigrations takes an ipfsRepoPath and newRepoPath
+// it creates a copy of the ipfs repo, moves it to the
+// new repo path and migrates that repo
+// it cleans up any tmp directories made, and removes
+// the new repo if any errors occur
+// IT DOES NOT REMOVE THE ORIGINAL REPO
+func RunMigrations(ipfsRepoPath, newRepoPath string) error {
+	// create temp directory into which we will copy the
+	// ipfs directory
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "ipfs_temp")
+	if err != nil {
+		return fmt.Errorf("error creating temp directory: %s", err)
+	}
+
+	rollback := func() {
+		if ipfsRepoPath != newRepoPath {
+			os.RemoveAll(newRepoPath)
```

Comment:
RemoveAll is usually pretty scary but don't have any input on how to make it better in a sane way.